client/app/bundles/course/video/components/tables/VideoTable.tsx

```diff
@@ -40,7 +40,22 @@ const VideoTable: FC<Props> = (props) => {
     return <Note message={intl.formatMessage(translations.noVideo)} />;
   }
 
-  videos.sort((a, b) => a.title.localeCompare(b.title));
+  const videoSortMethodByDateThenTitle = (val1, val2) => {
+    const post1 = val1 as VideoListData['startTimeInfo'];
+    const post2 = val2 as VideoListData['startTimeInfo'];
+
+    const date1 = new Date(post1.referenceTime!);
+    const date2 = new Date(post2.referenceTime!);
+    const dateComparison = date1.getTime() - date2.getTime();
+
+    if (dateComparison !== 0) {
+      return dateComparison;
+    } else {
+      return val1.title.localeCompare(val2.title);
+    }
+  };
+
+  const sortedVideos = videos.sort(videoSortMethodByDateThenTitle);
```

Comment:
Do we need to reassign the videos? The sorting in javascript is already done in-place as well.

Reply:
yes it's not needed. At first I just found the code a bit ugly, but I guess I can remove the re-assignment and simply do the sorting of the videos in-place